### PR TITLE
make contributor landing page performant

### PR DESCRIPTION
**Pull request by owocki**
I'm on a never ending quest to improve site performance.  When the application itself is performant, it allows us to spend LESS on app servers, and also to have a faster site.  Win win.

Specifically, the contributor landing page is slow.  This PR gets it down from 0.7s to 0.3s.

```
kevinowocki@local /Users/kevinowocki/Sites/gitcoin/web~ % time curl "https://gitcoin.co/contributor" > /dev/null
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 81744  100 81744    0     0   107k      0 --:--:-- --:--:-- --:--:--  107k
curl "https://gitcoin.co/contributor" > /dev/null  0.02s user 0.01s system 4% cpu 0.753 total
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 73081  100 73081    0     0   257k      0 --:--:-- --:--:-- --:--:--  258k
curl "http://localhost:8000/contributor" > /dev/null  0.00s user 0.00s system 3% cpu 0.285 total

```

The reason that the contributor landing page is slow was activity.view_props has ballooned.  To solve this, I implemented a caching layer on top of that information (`JSONStore`), and also made an investment in making all of the info in our database serializable (this was the bulk of the work).

As part of this investment, I was also able to build some infrastructure such that any object in our DB can/should be serializable.

### CODE REVIEW

File: app/perftools/management/commands/create_page_cache.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    print('results')
+    keywords = [''] + programming_languages
+    if settings.DEBUG:
+        keywords = ['']
+    view = 'results'
+    with transaction.atomic():
+        items = []
+        JSONStore.objects.filter(view=view).all().delete()
+        for keyword in keywords:
+            key = keyword
```
Comment (danlipert) [inline app/perftools/management/commands/create_page_cache.py]: Why are you copying the `keyword` here?

File: app/retail/views.py [hunk 2]
```diff
@@ -228,26 +228,42 @@ def contributor_bounties(request, tech_stack):
...
+        'projects': projects,
+    }
+
+    try:
+        new_context = JSONStore.objects.get(view='contributor_landing_page', key=tech_stack).data
+
+        for key, value in new_context.items():
+            context[key] = value
+    except Exception as e:
+        print(e)
```
Comment (danlipert) [inline app/retail/views.py]: Should we `print` here or log?

File: app/retail/views.py [hunk 3]
```diff
@@ -228,26 +228,42 @@ def contributor_bounties(request, tech_stack):
...
+    return TemplateResponse(request, 'bounties/contributor.html', context)
+
+
+def get_contributor_landing_page_context(tech_stack):
+    available_bounties_count = open_bounties().count()
+    available_bounties_worth = amount_usdt_open_work()
+    activities = get_activities(tech_stack)
+    return {
+        'activities': activities,
+        'title': tech_stack.title() + str(_(" Open Source Opportunities")) if tech_stack else "Open Source Opportunities",
```
Comment (danlipert) [inline app/retail/views.py]: Should we localize the else case here with `_()`?

**Comment by owocki** [comment]
@danlipert i turned around ur code review!

**Pull request merged**